### PR TITLE
Autogenerate markdown docs for kubectl

### LINKED ISSUE #3379: Autogenerate docs/cli.md from kubectl source
Manually updated docs have a tendency to drift from actual implementation. It shouldn't be too hard to add a script to generate the markdown from a [pkg/kubectl/cmd/cmd.go NewFactory](https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/kubectl/cmd/cmd.go#L68). We could then run the script manually or as a commit hook when changes are made to pkg/kubectl.

### PR DESCRIPTION
**Pull request by j3ffml**
Add test to verify generated docs are up to date. closes #3379

### CODE REVIEW

**Comment by jbeda** [comment]
How is this getting built?  I'd expect to see something under `build` someplace...

We should make sure that we don't include this binary with our binary releases.  Those are already too large and adding another binary will make them huge.  I also worry that this'll slow down the release build if we build it for every platform.

Did you consider making this a subcommand for kubectl instead?

**Comment by jlowdermilk** [comment]
> How is this getting built? I'd expect to see something under build someplace...

Added to golang.sh under `KUBE_TEST_TARGETS`.

> Did you consider making this a subcommand for kubectl instead?

I did consider that, but it seemed a bit strange/meta to include a doc generation command for kubectl in the command itself. It's also not particularly useful for end users, as if you already have the client you can just use --help.

**Comment by bgrant0607** [comment]
@nikhiljindal: We want to do something similar for API docs.

**Comment by lavalamp** [comment]
@filbranden for bash

**Comment by jlowdermilk** [comment]
Still having trouble getting `hack/verify-gendocs.sh` to work on travis (runs fine locally). @filbranden, bash-fu help appreciated.

**Comment by jbeda** [comment]
LGTM -- obviously we want to figure out the travis issues though.

File: hack/verify-gendocs.sh [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    echo "'./hack/build-go.sh cmd/gendocs'."
+  } >&2
+  exit 1
+fi
+
+
+KUBECTL_DOC="docs/kubectl.md"
+
+echo "diffing ${KUBECTL_DOC} against generated output from ${gendocs}"
+"${gendocs}" | diff ${KUBE_ROOT}/${KUBECTL_DOC} - && echo "${KUBECTL_DOC} up to date." || {
```
Comment (filbranden) [inline hack/verify-gendocs.sh]: Quotes around `"${KUBE_ROOT}/${KUBECTL_DOC}"`

**Comment by filbranden** [comment]
I have a small quote nitpick, otherwise shell scripts look good, feel free to proceed.

**Comment by jlowdermilk** [comment]
Fixed travis issues and rebased.

**Pull request merged**